src/tools/tidy/src/features.rs

```diff
@@ -187,6 +211,17 @@ pub fn collect_lang_features(base_src_path: &Path, bad: &mut bool) -> Features {
                 return None;
             }
 
+            if line.starts_with(FEATURE_GROUP_START_PREFIX) {
```

Comment:
To make this more robust, you could check whether `next_feature_group.is_some()` in here and `tidy_error!(..)` in that case so you can't start a new group in a group.

Reply:
Done.